DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/port_isolation/test_port_isolation_basic_functionality.py

```diff
@@ -0,0 +1,172 @@
+                "ip_destination": dev_groups[tg_ports[3-x]][0]["name"],
+                "srcMac": "28:be:0d:47:eb:2b",
+                "dstMac": "1c:99:9f:fb:63:15",
+                "frameSize": 162,
+                "rate": pps_value,
+                "protocol": "0x0800",
+                "type": "raw",
+            },
+            "stream1": {
+                "ip_source": dev_groups[tg_ports[1]][0]["name"],
+                "ip_destination": dev_groups[tg_ports[(3-x) if x<2 else 0]][0]["name"],
+                "srcIp": "147.147.96.74",
+                "dstIp": "229.112.223.59",
+                "srcMac": "f2:de:a4:35:bd:4b",
+                "dstMac": "01:00:5E:70:df:3b",
+                "frameSize": 162,
+                "rate": pps_value,
+                "protocol": "0x0800",
+                "type": "raw",
+            },
+            "stream2": {
+                "ip_source": dev_groups[tg_ports[2]][0]["name"],
+                "ip_destination": dev_groups[tg_ports[3-x if x <= 0 else 2-x]][0]["name"],
+                "srcMac": "2a:3e:13:88:e5:6d",
+                "dstMac": "ff:ff:ff:ff:ff:ff",
+                "frameSize": 162,
+                "rate": pps_value,
+                "protocol": "0x0800",
+                "type": "raw",
+            },
+            "stream3": {
+                "ip_source": dev_groups[tg_ports[3]][0]["name"],
+                "ip_destination": dev_groups[tg_ports[2-x]][0]["name"],
+                "srcMac": "ea:f9:ca:10:1d:b6",
+                "dstMac": "ba:2e:c5:2c:fa:8d",
+                "frameSize": 162,
+                "rate": pps_value,
+                "protocol": "0x0800",
+                "type": "raw",
+            }
+        }
+
+        await tgen_utils_setup_streams(tgen_dev, config_file_name=None, streams=streams)
+
+        await tgen_utils_start_traffic(tgen_dev)
+        await asyncio.sleep(traffic_duration)
+        await tgen_utils_stop_traffic(tgen_dev)
+
+        if x == 0:
+            # check the traffic stats
+            stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+            expected_loss = {
+                "stream0": 0,
+                "stream1": 0,
+                "stream2": 0,
+                "stream3": 0,
+            }
+            for row in stats.Rows:
+                assert tgen_utils_get_loss(row) == expected_loss[row["Traffic Item"]], \
+                    "Verify that traffic is forwarded in accordance."
+        else:
+            # check the traffic stats
+            stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+            expected_loss = {
+                "stream0": 100,
+                "stream1": 100,
+                "stream2": 100,
+                "stream3": 0,
+            }
+            for row in stats.Rows:
+                assert tgen_utils_get_loss(row) == expected_loss[row["Traffic Item"]], \
+                    "Verify that traffic is forwarded/not forwarded in accordance."
```

Comment:
Only the dict is changing. Everything else can be moved outside the `if` block:
```suggestion
        if x == 0:
            expected_loss = {
                "stream0": 0,
                "stream1": 0,
                "stream2": 0,
                "stream3": 0,
            }
        else:
            expected_loss = {
                "stream0": 100,
                "stream1": 100,
                "stream2": 100,
                "stream3": 0,
            }
        # check the traffic stats
        stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
        for row in stats.Rows:
            assert tgen_utils_get_loss(row) == expected_loss[row["Traffic Item"]], \
                "Verify that traffic is  forwarded/not forwarded in accordance."
```